Gave the Name Generator some love

Taorluath opened the pull request:
Behold, I have answered your cries! 
The big changes are:

Seraphim names are generated using Sheppard's propsed Seraphim vowel/consonant masher,  instead of a manually compiled list of names. The names generated by the new "makeSeraphimName()" method always seem to come out really well - this is much easier than [what originally looked like] an imagined list.

Aeon names are hard to find. It looked like you initially used single Aeon names (without surnames), which would have a very high chance of duplicates. I thought it would be a good idea to distinguish them by adding an epithet to the end of their names, thus adding diversity and role-play value. Of all changes, this might need some work (since depending on the words in the array, the generated name occasionally sounds awkward). However, the output of the Aeon names look very good to me overall, and I would be satisfied with its current state. 

UEF and Cybran surname lists have been considerably increased. Both the UEF and Cybran name and surname pools all have nearly 200 names each. The Cybran machine-name list likely needs more additions but I think it would be okay to have duplicates like"Hex-8" and "Hex-4" both appear in the same GW rotation. Also, a lot of people seem to think that the Cybrans have a significant German influence. I've added German names to the symbiont name pool and think they fit well with Cybran 'culture'. This also makes it easier to grow the name pool.

Trying to make your life easier,
(ShlippyShlappySlop)

Taorluath committed:
Gave the Name Generator some love

Behold, I have answered your cries! 
The big changes are:

Seraphim names are generated using Sheppard's propsed Seraphim vowel/consonant masher,  instead of a manually compiled list of names. The names generated by the new "makeSeraphimName()" method always seem to come out really well - this is much easier than [what originally looked like] an imagined list.

Aeon names are hard to find. It looked like you initially used single Aeon names (without surnames), which would have a very high chance of duplicates. I thought it would be a good idea to distinguish them by adding an epithet to the end of their names, thus adding diversity and role-play value. Of all changes, this might need some work (since depending on the words in the array, the generated name occasionally sounds awkward). However, the output of the Aeon names look very good to me overall, and I would be satisfied with its current state. 

UEF and Cybran surname lists have been considerably increased. Both the UEF and Cybran name and surname pools all have nearly 200 names each. The Cybran machine-name list likely needs more additions but I think it would be okay to have duplicates like"Hex-8" and "Hex-4" both appear in the same GW rotation. Also, a lot of people seem to think that the Cybrans have a significant German influence. I've added German names to the symbiont name pool and think they fit well with Cybran 'culture'. This also makes it easier to grow the name pool.

Trying to make your life easier,
(ShlippyShlappySlop)

Taorluath committed:
Noob Error

Fixed semi-colon issue on line 162.

Taorluath committed:
Update CharacterNameGenerator.java

Taorluath committed:
Update CharacterNameGenerator.java

Taorluath commented:
Apologize for the extra commits, realized I accidentally caused a type mismatch problem at first. Still learning to GitHub :)

Brutus5000 commented:
HI Taorluath,

thank you very much for your contribution. It may take a while until I can look over it, so please do not wonder if the PR remains open for a few days.

Regarding your commit issues:
You can merge unwanted commits with an interactive rebase like `git rebase -i HEAD~4` and squash the commits into the previous one (there are other possibilites too, see [github article](https://help.github.com/articles/about-git-rebase/) )
After that you need to force push the branch.

In this special case it is not required becauseI will merge all of your commits into one.

Brutus5000 commented:
@Taorluath I have a feature request to your PR:
Can you blacklist existing names from the SupCom campaign?

Taorluath committed:
Added a name blacklist

-Added a new array of names and surnames from SupCom campaigns that shouldn't appear in GW
-Added checks in the generateName() function to make sure the function returns a name that's not on the blacklist
-Minor changes: added and removed a few appropriate names from the name pools

Taorluath committed:
Compilation typo fix

Taorluath committed:
Variable name fix

Taorluath committed:
Scope error fix

Brutus5000 commented:
@Taorluath Thank you for this PR. If you have any more improvements, just open a new one.
